packages/shared/lib/services/nango-config.service.ts

```diff
@@ -299,240 +296,343 @@ export function convertV2ConfigObject(config: NangoConfigV2, showMessages = fals
+        if (sync.id) {
+            syncObject.id = sync.id;
+        }
 
-                    if (!allModelNames.includes(modelName as string)) {
-                        throw new Error(`Model ${modelName} not found included in models definition`);
-                    }
-                }
-                const modelFields = getFieldsForModel(action.input, config) as { name: string; type: string }[];
-                inputModel.name = action.input;
-                inputModel.fields = modelFields;
-            }
+        builtSyncs.push(syncObject);
+    }
 
-            let endpoints: NangoSyncEndpoint[] = [];
-            let actionEndpoint: string;
+    return { success: true, error: null, response: builtSyncs };
+}
 
-            if (action?.endpoint) {
-                if (Array.isArray(action?.endpoint)) {
-                    if (action?.endpoint?.length > 1) {
-                        const error = new NangoError('action_single_endpoint', actionName);
+function buildActions({
+    actions,
+    allModelNames,
+    config,
+    providerConfigKey,
+    showMessages,
+    isPublic,
+    allModels,
+    allEndpoints
+}: {
+    actions: NangoV2Integration;
+    allModelNames: string[];
+    config: NangoConfigV2;
+    providerConfigKey: string;
+    showMessages: boolean;
+    isPublic: boolean | null | undefined;
+    allModels: string[];
+    allEndpoints: string[];
+}): ServiceResponse<NangoSyncConfig[]> {
+    const builtActions: NangoSyncConfig[] = [];
+
+    for (const actionName in actions) {
+        const action: NangoIntegrationDataV2 = actions[actionName] as NangoIntegrationDataV2;
+        const {
+            success: modelSuccess,
+            error: modelError,
+            response: models
+        } = formModelOutput({ integrationData: action, allModels, allModelNames, config, name: actionName, type: 'action' });
+
+        if (!modelSuccess || !models) {
+            return { success: false, error: modelError, response: null };
+        }
 
-                        return { success: false, error, response: null };
-                    }
-                    actionEndpoint = action?.endpoint[0] as string;
-                } else {
-                    actionEndpoint = action?.endpoint;
-                }
+        let inputModel: NangoSyncModel = {} as NangoSyncModel;
 
-                endpoints = assignEndpoints(actionEndpoint, 'POST', false, showMessages);
-                if (actionEndpoint?.includes('{') && actionEndpoint.includes('}')) {
-                    const { success, error, response } = parseModelInEndpoint(actionEndpoint, allModelNames, inputModel, config);
-                    if (!success || !response) {
-                        return { success, error, response: null };
-                    }
-                    inputModel = response;
-                }
+        if (action.input) {
+            if (action.input.includes('{') && action.input.includes('}')) {
```

Comment:
you can use the optional chaining operator and get rid of the first `if` I think: `action.input?...`